packages/convex-helpers/testing.ts

```diff
@@ -0,0 +1,102 @@
+import { ConvexClient } from "convex/browser";
+import {
+  FunctionArgs,
+  FunctionReference,
+  FunctionReturnType,
+  UserIdentity,
+} from "convex/server";
+
+/**
+ * This is a helper for testing Convex functions against a locally running backend.
+ *
+ * An example of calling a function:
+ * ```
+ * const t = new ConvexTestingHelper();
+ * const result = await t.query(api.foo.bar, { arg1: "baz" })
+ * ```
+ *
+ * An example of calling a function with auth:
+ * ```
+ * const t = new ConvexTestingHelper();
+ * const identityA = t.newIdentity({ name: "Person A"})
+ * const result = await t.withIdentity(identityA).query(api.users.getProfile);
+ * ```
+ */
+export class ConvexTestingHelper {
+  private _nextSubjectId: number = 0;
+  public client: ConvexClient;
+  private _adminKey: string;
+
+  constructor(options: { adminKey?: string; backendUrl?: string } = {}) {
+    this.client = new ConvexClient(
+      options.backendUrl ?? "http://127.0.0.1:8000"
+    );
+    this._adminKey =
+      options.adminKey ??
+      // default admin key for local backends - from https://github.com/get-convex/convex-backend/blob/main/Justfile
+      "0135d8598650f8f5cb0f30c34ec2e2bb62793bc28717c8eb6fb577996d50be5f4281b59181095065c5d0f86a2c31ddbe9b597ec62b47ded69782cd"
+  }
+
+  newIdentity(
+    args: Partial<Omit<UserIdentity, "tokenIdentifier">>
+  ): Omit<UserIdentity, "tokenIdentifier"> {
+    const subject = `test subject ${this._nextSubjectId}`;
+    this._nextSubjectId += 1;
+    const issuer = "test issuer";
+    return {
+      ...args,
+      subject,
+      issuer,
+    };
+  }
+
+  withIdentity(
+    identity: Omit<UserIdentity, "tokenIdentifier">
+  ): Pick<ConvexClient, "mutation" | "action" | "query"> {
+    return {
+      mutation: (functionReference, args) => {
+        (this.client as any).setAdminAuth(this._adminKey, identity);
+        return this.client.mutation(functionReference, args).finally(() => {
+          this.client.client.clearAuth();
```

Comment:
is there interleaving edge case behavior if you're running multiple mutations in parallel during a test? Or will the synchronous setAdmin, then start the call, mostly be fine? Worst case you clear someone else's auth or something

Reply:
Yeah I actually want to switch this to using the HTTP client so it's less stateful, but the change to add the second argument to `setAdminAuth` requires an npm release (but I landed the change already)